File: bad-tests/parallel_construct_wait_clause.cpp
#include "acc_testsuite.h"

#ifndef T1
/*T1:parallel construct wait clause,V:2.0-2.7*/
int test1(){
    int err = 0;
    srand(SEED);
    const int array_size = 1024;
    int *a = (int*)malloc(array_size * sizeof(int));
    int *b = (int*)malloc(array_size * sizeof(int));
   
    // Initialize a and b arrays with random data
    for(int i = 0; i < array_size; i++) {
        a[i] = rand() % 1024;
        b[i] = 0;
    }
    
    #pragma acc data copyin(a[0:array_size]) copyout(b[0:array_size])
    {
        #pragma acc parallel async
        {
            #pragma acc loop
            for(int i = 0; i < array_size; i++) {
                b[i] = a[i] * 2;
            }
        }

        #pragma acc wait

        for(int i = 0; i < array_size; i++) {
            if(b[i] != a[i] * 2) {
                err = 1;
                break;
            }
        }
    }

    free(a);
    free(b);

    return err;
}
#endif

int main(){
    int failcode = 0;
    int failed;
#ifndef T1
    failed = 0;
    for (int x = 0; x < NUM_TEST_CALLS; ++x){
        failed = failed + test1();
    }
    if (failed != 0){
        failcode = failcode + (1 << 0);
    }
#endif
    return failcode;
}

File: Phind-CodeLlama-34B-v2/context-template-full/parallel_construct_wait_clause.f90
#ifndef T1
!T1:parallel construct wait clause,V:2.7-2.3
      LOGICAL FUNCTION test1()
        USE OPENACC
        IMPLICIT NONE
        INCLUDE "acc_testsuite.Fh"
        
        INTEGER :: errors = 0
        INTEGER, DIMENSION(10) :: a, b, c
        INTEGER :: i

        DO i = 1, 10
          a(i) = i
          b(i) = i * 2
        END DO

        !$acc parallel copy(a(1:10), b(1:10)) async(1)
          !$acc loop
          DO i = 1, 10
            a(i) = a(i) + b(i)
          END DO
        !$acc end parallel

        !$acc parallel copy(a(1:10)) async(2)
          !$acc loop
          DO i = 1, 10
            a(i) = a(i) * 2
          END DO
        !$acc end parallel

        !$acc parallel wait(1) async(3)
          !$acc loop
          DO i = 1, 10
            a(i) = a(i) - 1
          END DO
        !$acc end parallel

        !$acc parallel wait(2, 3) async(4)
          !$acc loop
          DO i = 1, 10
            a(i) = a(i) * 3
          END DO
        !$acc end parallel

        !$acc parallel wait(4) async(5)
          !$acc loop
          DO i = 1, 10
            a(i) = a(i) + 1
          END DO
        !$acc end parallel

        !$acc parallel wait(5) async(6)
          !$acc loop
          DO i = 1, 10
            a(i) = a(i) - 2
          END DO
        !$acc end parallel

        !$acc parallel wait(6) async(7)
          !$acc loop
          DO i = 1, 10
            a(i) = a(i) * 4
          END DO
        !$acc end parallel

        !$acc parallel wait(7) async(8)
          !$acc loop
          DO i = 1, 10
            a(i) = a(i) + 3
          END DO
        !$acc end parallel

        !$acc parallel wait(8) async(9)
          !$acc loop
          DO i = 1, 10
            a(i) = a(i) - 4
          END DO
        !$acc end parallel

        !$acc parallel wait(9) async(10)
          !$acc loop
          DO i = 1, 10
            a(i) = a(i) * 5
          END DO
        !$acc end parallel

        DO i = 1, 10
          IF (a(i) .ne. ((i + i * 2 - 1 + 3) * 5)) THEN
            errors = errors + 1
          END IF
        END DO

        IF (errors .eq. 0) THEN
          test1 = .FALSE.
        ELSE
          test1 = .TRUE.
        END IF
      END
#endif

      PROGRAM main
        IMPLICIT NONE
        INTEGER :: failcode, testrun
        LOGICAL :: failed
        INCLUDE "acc_testsuite.Fh"
#ifndef T1
        LOGICAL :: test1
#endif
        failed = .FALSE.
        failcode = 0
#ifndef T1
        DO testrun = 1, NUM_TEST_CALLS
          failed = failed .or. test1()
        END DO
        IF (failed) THEN
          failcode = failcode + 2 ** 0
          failed = .FALSE.
        END IF
#endif
        CALL EXIT (failcode)
      END PROGRAM